### DISCUSSION TITLE
Component Select when column nullable

### DISCUSSION THREAD

**Discussion by obeidareda37** [filamentphp/filament#13517] [Help] [answered]
### Package

Form builder

### Package Version

v3.2

### When using the Select component in a form and trying to fetch values from a relationship, I face some challenges. Specifically, when there is any empty record in the relationship table, it throws an error. However, if all records have values, it works fine.

**The first select works fine because  all record columns in the model have a value 
The second select return error because there are a nullable column in model** 

**This is what the error looks like**

![Screenshot 2024-07-09 at 9 07 24 PM](https://github.com/filamentphp/filament/assets/70585602/724cc5c4-6536-46a4-8664-ddf8a95bfbd4)

**This form in filament resource**

<img width="798" alt="Screenshot 2024-07-09 at 9 12 34 PM" src="https://github.com/filamentphp/filament/assets/70585602/888d6b7d-b744-4e5d-ab68-2424f3bfc5e6">

**and This relation** 

<img width="760" alt="Screenshot 2024-07-09 at 9 13 06 PM" src="https://github.com/filamentphp/filament/assets/70585602/f8bca142-ef9d-457a-a8f3-8945192cea30">

any solution or work around ?

**Comment by obeidareda37** [comment]
@danharrin

**Reply by leandrocfe** [reply, marked as answer]
```php
Select::make('visitor.vehicle_plate_number')
    ->relationship(
        name: 'visitor',
        titleAttribute: 'vehicle_plate_number',
        modifyQueryUsing: fn (Builder $query) => $query->whereNotNull('vehicle_plate_number')
    )
```
?

https://filamentphp.com/docs/3.x/forms/fields/select#customizing-the-relationship-query

**Reply by obeidareda37** [reply]
But is it possible to make it receive the value of null without resorting to modifyQueryUsing 
I did this solution and uploaded it to PR
https://github.com/filamentphp/filament/pull/13520

this correct or not?

@danharrin

**Reply by danharrin** [reply]
How would the user know what option they are selecting if there is no label? That is why it is correct to filter out all options without a label

**Comment by obeidareda37** [comment]
i open PR for this problem
https://github.com/filamentphp/filament/pull/13520
